File: app/src/main/java/in/smslite/viewHolder/SearchSmsViewHolder.java
package in.smslite.viewHolder;

import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.net.Uri;
import android.provider.ContactsContract;
import android.support.v7.widget.RecyclerView;
import android.text.Spanned;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import in.smslite.R;
import in.smslite.activity.CompleteSmsActivity;
import in.smslite.contacts.Contact;
import in.smslite.contacts.PhoneContact;
import in.smslite.utils.ContactUtils;
import in.smslite.utils.TimeUtils;

import static in.smslite.activity.MainActivity.db;

/**
 * Created by rahul1993 on 4/22/2018.
 */

public class SearchSmsViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
  private static final String TAG = SearchSmsViewHolder.class.getSimpleName();
  private View view;
  private String address;
  private Context mContext;
  private TextView summaryView;
  private TextView titleView;
  private TextView timeView;
  private ImageView imageView;

  public SearchSmsViewHolder(View itemView, Context context) {
    super(itemView);
    this.mContext = context;
    view = itemView;
    titleView = (TextView) view.findViewById(R.id.sms_title);
    summaryView = (TextView) view.findViewById(R.id.sms_summary);
    timeView = (TextView) view.findViewById(R.id.sms_time);
    imageView = (ImageView) view.findViewById(R.id.avatar);
    view.setOnClickListener(this);
  }

  public void setAddress(String address) {
    if (address == null || address.isEmpty()) {
      address = "Unknown Sender";
    }
    this.address = address;
    Contact contact = ContactUtils.getContact(address, mContext, true);
    titleView.setText(contact.getDisplayName());
    setAvatar(contact);
  }

  public void setSummary(Spanned smsBody) {
    summaryView.setText(smsBody);
  }

  public void setTime(long x) {
    String time = TimeUtils.getPrettyElapsedTime(x);
    timeView.setText(time);
  }

  @Override
  public void onClick(View view) {
    Intent i = new Intent(mContext, CompleteSmsActivity.class);
    i.putExtra(view.getResources().getString(R.string.address_id), address);
    mContext.startActivity(i);
  }

  private void setAvatar(Contact contact) {
    Drawable drawable = contact.getAvatar(mContext);
    imageView.setImageDrawable(drawable);
    imageView.setOnClickListener(onImageClick(contact));
  }

  private View.OnClickListener onImageClick(final Contact contact) {
    return new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        if (Contact.Source.PHONE.equals(contact.getSource())) {
          PhoneContact phoneContact = (PhoneContact) contact;
          ContactsContract.QuickContact.showQuickContact(mContext, view,
              phoneContact.getUri(),
              ContactsContract.QuickContact.MODE_LARGE, null);
        } else {
          Intent intent = new Intent(ContactsContract.Intents.SHOW_OR_CREATE_CONTACT,
              Uri.fromParts("tel", address, null));
          v.getContext().startActivity(intent);
        }
      }
    };
  }
}

File: app/src/main/java/in/smslite/viewHolder/SearchSmsViewHolder.kt
package `in`.smslite.viewHolder

import `in`.smslite.R
import `in`.smslite.activity.CompleteSmsActivity
import `in`.smslite.contacts.Contact
import `in`.smslite.contacts.PhoneContact
import `in`.smslite.utils.ContactUtils
import `in`.smslite.utils.TimeUtils
import android.content.Context
import android.content.Intent
import android.graphics.drawable.Drawable
import android.net.Uri
import android.provider.ContactsContract
import android.text.Spanned
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by rahul1993 on 4/22/2018.
 */
class SearchSmsViewHolder(private val view: View, private val mContext: Context) : RecyclerView.ViewHolder(view), View.OnClickListener {
    private var address: String? = null
    private val summaryView: TextView
    private val titleView: TextView
    private val timeView: TextView
    private val imageView: ImageView
    fun setAddress(address: String?) {
        var address = address
        if (address == null || address.isEmpty()) {
            address = "Unknown Sender"
        }
        this.address = address
        val contact: Contact? = ContactUtils.getContact(address, true)
        titleView.text = contact?.displayName
        if (contact != null) {
            setAvatar(contact)
        }
    }

    fun setSummary(smsBody: Spanned?) {
        summaryView.text = smsBody
    }

    fun setTime(x: Long) {
        val time = TimeUtils.getPrettyElapsedTime(x)
        timeView.text = time
    }

    override fun onClick(view: View) {
        val i = Intent(mContext, CompleteSmsActivity::class.java)
        i.putExtra(view.resources.getString(R.string.address_id), address)
        mContext.startActivity(i)
    }

    private fun setAvatar(contact: Contact) {
        val drawable: Drawable = contact.getAvatar(mContext)
        imageView.setImageDrawable(drawable)
        imageView.setOnClickListener(onImageClick(contact))
    }

    private fun onImageClick(contact: Contact): View.OnClickListener {
        return View.OnClickListener { v ->
            if (Contact.Source.PHONE == contact.source) {
                val phoneContact: PhoneContact = contact as PhoneContact
                ContactsContract.QuickContact.showQuickContact(mContext, view,
                        phoneContact.uri,
                        ContactsContract.QuickContact.MODE_LARGE, null)
            } else {
                val intent = Intent(ContactsContract.Intents.SHOW_OR_CREATE_CONTACT,
                        Uri.fromParts("tel", address, null))
                v.context.startActivity(intent)
            }
        }
    }

    companion object {
        private val TAG = SearchSmsViewHolder::class.java.simpleName
    }

    init {
        titleView = view.findViewById<View>(R.id.sms_title) as TextView
        summaryView = view.findViewById<View>(R.id.sms_summary) as TextView
        timeView = view.findViewById<View>(R.id.sms_time) as TextView
        imageView = view.findViewById<View>(R.id.avatar) as ImageView
        view.setOnClickListener(this)
    }
}